.pre-commit-config.yaml

```
@@ -39,6 +48,21 @@ repos:
           - license-templates/LICENSE.txt
         files: >
           \.properties$|\.cfg$|\.conf$|\.ini$|\.ldif$|\.readthedocs$|\.service$|^Dockerfile.*$
+      - id: insert-license
+        name: Add licence for all JS files
+        files: \.js$
+        exclude: ^\.github/.*$|^airflow/_vendor/.*$
+        args:
+          - --comment-style
+          - "/**| *| */"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+      - id: debug-statements
+      - id: fix-encoding-pragma
+        exclude: ^\.github/.*$"|^airflow/_vendor/.*$
```

```suggestion
        exclude: ^\.github/.*$"|^airflow/_vendor/.*$
        args:
          - --remove
```
We could also remove them for Airflow 2.0. What do you think?

---

Aaaah. I did not realise we do not need them in python 3. Yes. --remove in this case would be nice. But then we touch ALL python files generating potentially lots and lots of conflicts and this will impose fixing all pylint errors for those files. I made sure to fix all the pylint errors in files touched by LICENCE/encoding pragma :).

I'd say let's do it as one single big commit after we finish pylint changes ?

---

Is it worth removing this stage so that we don't introduce changes in half the files that we then back out in all the files later?

---

Right. Makes perfect sense @ashb :)

---

I removed all the pragma changes and split this change in two - one adding consistent licences and one introducing related pylint changes in affected files (this way it will be easier to cherry-pick only the licence changes in 1.10.5).